uefi/src/main.rs

```diff
@@ -0,0 +1,441 @@
+#![no_std]
+#![no_main]
+#![feature(abi_efiapi)]
+#![deny(unsafe_op_in_unsafe_fn)]
+
+use crate::memory_descriptor::UefiMemoryDescriptor;
+use bootloader_api::{info::FrameBufferInfo, BootloaderConfig};
+use bootloader_x86_64_common::{legacy_memory_region::LegacyFrameAllocator, Kernel, SystemInfo};
+use core::{arch::asm, cell::UnsafeCell, fmt::Write, mem, panic::PanicInfo, ptr, slice};
+use uefi::{
+    prelude::{entry, Boot, Handle, Status, SystemTable},
+    proto::{
+        console::gop::{GraphicsOutput, PixelFormat},
+        device_path::DevicePath,
+        loaded_image::LoadedImage,
+        media::{
+            file::{File, FileAttribute, FileInfo, FileMode},
+            fs::SimpleFileSystem,
+        },
+        network::{
+            pxe::{BaseCode, DhcpV4Packet},
+            IpAddress,
+        },
+    },
+    table::boot::{
+        AllocateType, MemoryDescriptor, MemoryType, OpenProtocolAttributes, OpenProtocolParams,
+    },
+    CStr16, CStr8,
+};
+use x86_64::{
+    structures::paging::{FrameAllocator, OffsetPageTable, PageTable, PhysFrame, Size4KiB},
+    PhysAddr, VirtAddr,
+};
+
+mod memory_descriptor;
+
+static SYSTEM_TABLE: VeryUnsafeCell<Option<SystemTable<Boot>>> = VeryUnsafeCell::new(None);
+
+struct VeryUnsafeCell<T>(UnsafeCell<T>);
+
+impl<T> VeryUnsafeCell<T> {
+    const fn new(v: T) -> Self {
+        Self(UnsafeCell::new(v))
+    }
+}
```

Comment:
This should probably renamed to `RacyCell` to be consistent with 
https://github.com/rust-osdev/bootloader/blob/7088b5344442aadbc138d247944ca34ffdccf4cc/bios/common/src/racy_cell.rs#L5